src/packet/signature/config.rs

```diff
@@ -299,11 +299,19 @@ impl SignatureConfig {
         }
     }
 
-    /// Returns an iterator over all subpackets of this signature.
-    pub fn subpackets(&self) -> impl Iterator<Item = &Subpacket> {
-        self.hashed_subpackets
-            .iter()
-            .chain(self.unhashed_subpackets.iter())
+    #[deprecated(note = "Use hashed_subpackets")]
+    pub fn subpackets(&self) -> impl Iterator<Item=&Subpacket> {
+        self.hashed_subpackets()
```

Comment:
the deprecation is a good idea, but we should keep the functionality as is in this case, otherwise this is still a breaking change

Reply:
My perspective is that the previous implementation was a bug, and that this PR shifts the semantics to what they should always have been. With the deprecation warning, as it currently stands, users of the API get notified and have a chance to reflect if they actually intended to read from the unhashed area.

But of course this is arguable.

And the question of how to deal with adjusting the subpacket getter semantics is even more tricky: Note that the semantics of subpacket getters in `types.rs` are changed by this PR.

However, while this is formally a "breaking change", I wouldn't consider this change "breaking".
Unfortunately, the RFC doesn't offer clear guidance about which subpacket types are reasonable to store in the unhashed area.

Reply:
Right, that's fair. I still worry about how applications might use these, but maybe that's not actionable in a reasonable way.

At this point, i am happy to drop this issue, for now. In my use of rpgp, I shouldn't use any of these accessors. The abstraction they provide is not right for my use case.

Maybe in the long run, the right thing to do is to point users at a higher level abstraction library, and to deprecate these subpacket access helper functions in rpgp?

Reply:
> Maybe in the long run, the right thing to do is to point users at a higher level abstraction library, and to deprecate these subpacket access helper functions in rpgp?

Probably for most use cases, I think it's good for lower level tools to have it though.

Reply:
Exposing low level access to subpackets from rpgp is reasonable, right. But then the access should expose the OpenPGP data without interpreting it.

Most importantly, it should not conflate the two subpacket areas - they are separated in the packet format precisely because their semantics are very different, even though their data format is of course identical between the two subpacket areas.

If we want to go in that direction (which I'd find worthwhile), I could propose a change to the API to expose uninterpreted access to the subpackets.

Reply:
yes that would be nice